deepspeed/runtime/compiler.py

```diff
@@ -0,0 +1,173 @@
+    Please be aware that these features and API designs are experimental and subject to change.
+    """
+
+    enabled: bool = False
+    """
+    Enable torch.compile when True.
+    """
+
+    backend: str = "inductor"
+    """
+    Passed to `backend` argument of torch.compile.
+    If the given value is not in torch._dynamo.list_backends(),
+    DeepSpeed attempts to import and instantiate the module with the given name.
+    """
+
+    kwargs: Dict[str, Any] = {}
+    """
+    Passed to `kwargs` argument of torch.compile.
+    """
+
+    @validator("enabled")
+    def validate_enabled(cls, field_value, values):
+        if field_value and not is_compile_supported():
+            raise ValueError("torch.compile is not supported on this version of PyTorch.")
+        return field_value
+
+    @validator("backend")
+    def validate_backend(cls, field_value, values):
+        if is_compile_supported():
+            if isinstance(field_value, str):
+                if field_value not in torch._dynamo.list_backends():
+                    return get_backend_fn(field_value)
+            elif not callable(field_value):
+                raise ValueError(f"backend for torch.compile must be a string or Callable: {field_value}")
+
+        return field_value
+
+
+class CompiledModuleWrapper(torch.nn.Module):
+
+    def __init__(self, module, compile_config: Union[CompileConfig, None] = None):
+        super().__init__()
+
+        assert is_compile_supported(), "torch.compile is not supported on this version of PyTorch."
+
+        modules = self.__dict__.get('_modules')
+        modules['wrapped'] = module
+        self.__dict__['wrapped'] = module
+        self._is_compiled = False
+        self._backend = compile_config.backend
+        self._compile_kwargs = compile_config.kwargs
+        self._compiler_fn = None
+
+    def __getattr__(self, name):
+        return getattr(self.__dict__['wrapped'], name)
+
+    def set_backend(self, backend: Union[str, Callable]):
+        """Set the backend for torch.compile.
+
+        Args:
+            backend (Union[str, Callable]): backend name or a function that takes a torch.nn.Module and returns a compiled module.
+            You can directly pass a function that works as a backend.
+            See also `backend` field in `CompileConfig` for more details.
+        """
+        if isinstance(backend, str):
+            self._backend = get_backend_fn(backend)
+        elif callable(backend):
+            self._backend = backend
+        else:
+            raise ValueError(f"backend for torch.compile must be a string or Callable: {backend}")
+
+    def set_torch_compile_kwargs(self, kwargs: Dict[str, Union[str, Any]]) -> None:
+        """Set kwargs for torch.compile. Kwargs that are set in DeepSpeed config will be overwritten.
+        You can also pass a backend name with "backend" key to change the backend.
```

Comment:
I think it is confusing to allow users to change backend through `set_torch_compile_kwargs`, especially since we `set_backend` already exists for that purpose.

Reply:
I see, you are right. We already have `set_backend`. Let me fix it.